torchinductor/codegen/triton.py

```diff
@@ -1098,10 +1097,21 @@ def end_current_reduction_loop():
         log.info("schedule: %s", node_schedule)
         return self.codegen_node_schedule(node_schedule, numel, rnumel)
 
+    @staticmethod
+    def is_contiguous(node):
+        if node in (EnableReduction, DisableReduction):
+            return True
+        return all(
+            dep.is_contiguous()
+            for dep in itertools.chain(node.read_writes.reads, node.read_writes.writes)
```

Comment:
should it depend on whether reduction is inner/outer, and not just on whether the dep is contiguous? Also, even for contiguous reductions we needed different autotuning configs (relying on just one resulted in 1.5x or so slower kernels)

Reply:
"contiguous" here means if the memory indexing formula is contiguous, not if the tensor is contiguous.  So I believe outer reduction memory deps will show up as `x0 + r0*stride`, and looks like transposed memory accesses with `is_contiguous()` being false.

I believe my change from 
```
triton_config_reduction(size_hints, 1, 2048)
```
to
```
triton_config_reduction(size_hints, 1, (rnumel if 256 <= rnumel < 2048 else 2048))
```
Fixed the issue where we needed to autotune different configs for contiguous kernels.  I actually saw some speedups from this change alone, and it improved the gmean perf on torchbench on my local GPU.

Is there a specific model or sizes I should check?

Reply:
I saw forward and backward kernels for layer norm requiring different configs (2x1024 vs 8 x 512 when rnumel was 1024) in hf torchbench models, but perf results show they are fine.